### DISCUSSION TITLE
Different ESP32 Cam Devices on the Market?

### DISCUSSION THREAD

**Discussion by ullibak** [jomjol/AI-on-the-edge-device#764] [Q&A] [answered]
I ordered a ESP32 Cam bundled with a plug-on USB-port programmer, like e.g. shown on the picture in
https://randomnerdtutorials.com/upload-code-esp32-cam-mb-usb/
I was able to download the firmware and the device seemed to run. But it was very unreliable. I got a lot of errors when loading new pages in the browser and thought that minor differences in files might be the cause, see issue https://github.com/jomjol/AI-on-the-edge-device/issues/736

I noticed the same behaviour on three different devices of this type. After trying a lot of things (WLAN, different browsers etc.), I ordered a device that was not bundled with a programmer and hat to be programmed as described in the Wiki using a USB-to-TTL adapter. With this device, set-up and confituration using the browser ran well and was stable. 

Both devices are labelled "ESP32-S"  on the processor module, but only the device that worked well hat the "AI" logo. Also, I noticed minor differences on the circuit board. The camera module is different, but this can hardly explain errors like "httpd_sock_err: error in send : 104"

Did anyone have a similar experience and perhaps knowledge of differences between similarly looking ESP32-Cam devices on the market? Here are pictures of both devices:
![ESP32-Cam](https://user-images.githubusercontent.com/26085758/162450111-88e4944c-a5ea-4637-a6e0-c87c979cccbd.jpg)

**Comment by pixeldoc2000** [comment, marked as answer]
Similar "looking" Board can have major differences:
- Processor
- Ram (Size! & Type) -&gt; this Project needs 4MB RAM!
- Flashrom
- Camera Modules
- Antenna
- Quality of Components
- Manufacture Quality of the PCB and soldering
- Different Components
- "Clone" Components -&gt; ESPxx
- etc.

This can cause different Power Consumption, Power Requirements, compatibility issues, etc.

Most manufacturers and sellers buy what's cheap today on the Shenzhen market, that's how it works...

In the end, it looks like it is a trial and error approach which ESP32-CAM Module works reliable.

**Reply by caco3** [reply]
added task to update wiki page, see https://github.com/jomjol/AI-on-the-edge-device/issues/1107

**Comment by RMadosa** [comment]
Hi ullibak
I compared my ESP32 and found that I don't have any problems with the ones that don't work for you. Again I bought another variant, which is not stable.
If I compare the MAC addresses, then there is a certain consistency.

Unfortunately, no matter which address I check, the same manufacturer always comes out. But if 3 out of 3 work stably with the same vendor code, then that's a good indication. I think so.

Is working:
Piece / Vendor
2 / 30:C6:F7 - Espressif Inc.
2 / 0C:B8:15 - Espressif Inc.
1 / EC:62:60 - Espressif Inc.
![ESP32 - Good](https://user-images.githubusercontent.com/48294024/194710343-129274fc-df1d-4660-b244-665ce1c434c0.jpg)

It does not work:
Piece / Vendor
2 / C0:49:EF - Espressif Inc.
1 / C4:DE:E2 - Espressif Inc.
![ESP32 - Bad](https://user-images.githubusercontent.com/48294024/194710367-1cddaef3-95df-4435-8cb8-082b0c0aa01b.jpg)

Piece = Number of tested ESP32
The fact that the USB adapters are different made no difference for me

Tested with these camera modules
Camera Type OMV2640 (cable length-degree)
75-66 (top), 21-120 (button left) and 21-66 (bottom right)   
![ESP - CAM - Good](https://user-images.githubusercontent.com/48294024/194710389-9cdf4c3c-91b9-4939-b4a5-acdae248291e.jpg)
